src/iam/iam-base.ts

```diff
@@ -242,6 +245,8 @@ export class IAMBase {
             this._provider = new providers.Web3Provider(wcProvider);
             this._signer = this._provider.getSigner();
             this._providerType = walletProvider;
+            const walletConnector = this._walletConnectService.getConnector();
+            this._walletConnectSignMessage = walletConnector.signMessage.bind(walletConnector);
```

Comment:
There a lot of places in code where we need signing. Wouldn't be better to implement Signer on top of wallet connect?